### PR TITLE
Airtable Connector

### PR DESCRIPTION
**Pull request by jman0119**

### CODE REVIEW

File: airbyte-integrations/connectors/source-airtable/oauth2.patch.json [hunk 1]
```diff
@@ -0,0 +1,8 @@
+{
+    "provider": "airtable",
+    "authUrlTemplate": "https://airtable.com/oauth2/v1/authorize?client_id={{#urlencode}}{{{ client_id }}}{{/urlencode}}&redirect_uri={{#urlencode}}{{{ redirect_uri }}}{{/urlencode}}&response_type=code&state={{#urlencode}}{{{ state }}}{{/urlencode}}&scope=data.records:read%20data.recordComments:read%20schema.bases:read    ",
```
Comment (mdibaiee) [inline airbyte-integrations/connectors/source-airtable/oauth2.patch.json:3]: extra spaces at the end
```suggestion
    "authUrlTemplate": "https://airtable.com/oauth2/v1/authorize?client_id={{#urlencode}}{{{ client_id }}}{{/urlencode}}&redirect_uri={{#urlencode}}{{{ redirect_uri }}}{{/urlencode}}&response_type=code&state={{#urlencode}}{{{ state }}}{{/urlencode}}&scope=data.records:read%20data.recordComments:read%20schema.bases:read",
```

File: airbyte-integrations/connectors/source-airtable/oauth2.patch.json [hunk 2]
```diff
@@ -0,0 +1,8 @@
...
+    "accessTokenUrlTemplate": "https://airtable.com/oauth2/v1/token",
+    "accessTokenBody": "grant_type=authorization_code&client_id={{#urlencode}}{{{ client_id }}}{{/urlencode}}&client_secret={{#urlencode}}{{{ client_secret }}}{{/urlencode}}&redirect_uri={{#urlencode}}{{{ redirect_uri }}}{{/urlencode}}&code={{#urlencode}}{{{ code }}}{{/urlencode}}",
```
Comment (mdibaiee) [inline airbyte-integrations/connectors/source-airtable/oauth2.patch.json:5]: According to [Airtabledocumentation]( https://airtable.com/developers/web/api/oauth-reference#token-creation-request) this request does not take `client_secret`, instead it takes a header (we would put that in `accessTokenHeaders`) with encodes both `client_id` and `client_secret`.

This is assuming we do have a `client_secret` for AirTable. Have we created an OAuth2 app on AirTable yet? That would help us decide how to implement the oauth2 flow here.

Comment (mdibaiee) [reply airbyte-integrations/connectors/source-airtable/oauth2.patch.json:5]: Apparently AirTable also has additional parameters [`code_challenge` and `code_verifier`](https://airtable.com/developers/web/api/oauth-reference#authorization-parameter-rules) which we will need to generate cryptographically and send (apparently there is an extension to OAuth2 protocol called PKCE which requires these).

This needs to be implemented in our flow, we currently don't support this PKCE extension as part of our OAuth flow. I will create an issue for this, needs work on front-end as well.

File: airbyte-integrations/connectors/source-airtable/spec.patch.json [hunk 3]
```diff
@@ -0,0 +1,38 @@
+{
+      "properties": {
+        "credentials": {
+          "title": "Authentication",
+          "type": "object",
+          "discriminator": {
+            "propertyName": "auth_method"
+          },
+          "oneOf": [
```
Comment (mdibaiee) [inline airbyte-integrations/connectors/source-airtable/spec.patch.json:9]: Arrays do not get merged by [JSON Merge Patch](https://www.rfc-editor.org/rfc/rfc7396.txt), and so *all* properties must be included when we are overwriting arrays. In this case we are missing `client_id`, `client_secret`, `refresh_token`, `api_key`, `access_token` and `token_expiry_date`

File: airbyte-integrations/connectors/source-airtable/oauth2.patch.json [hunk 4]
```diff
@@ -0,0 +1,8 @@
...
+    "accessTokenHeaders": {"content-type": "application/x-www-form-urlencoded"},
+    "accessTokenResponseMap": {"refresh_token": "/refresh_token"}
```
Comment (mdibaiee) [inline airbyte-integrations/connectors/source-airtable/oauth2.patch.json:7]: The response includes more fields. If you look at `spec.json` of this connector, we see three fields that the connector expects to receive from the endpoint: `access_token`, `refresh_token` and `token_expires_in`. In the [AirTable docs](https://airtable.com/developers/web/api/oauth-reference#token-creation-request) you can find the original names of these fields. In this case, we would have:

```
{
  "refresh_token": "/refresh_token",
  "access_token": "/access_token",
  "token_expires_in": "/expires_in"
}
```

File: airbyte-integrations/connectors/source-airtable/oauth2.patch.json (hunk 2 above)
Comment (jman0119) [reply airbyte-integrations/connectors/source-airtable/oauth2.patch.json:5]: @mdibaiee yes I created an app on Airtable so we have a client_id and client_secret that we can use.

File: airbyte-integrations/connectors/source-airtable/oauth2.patch.json (hunk 4 above)
Comment (jman0119) [reply airbyte-integrations/connectors/source-airtable/oauth2.patch.json:7]: @mdibaiee when I searched spec.json, I don't see `token_expires_in`, are you referring to `token_expiry_date`, or am I not looking at the right place?

File: airbyte-integrations/connectors/source-airtable/spec.patch.json (hunk 3 above)
Comment (jman0119) [reply airbyte-integrations/connectors/source-airtable/spec.patch.json:9]: @mdibaiee Is everything in the `oneOf` considered part of the array? If so, the `api_key` in the Personal Access Token section also is part of the `oneOf` array? Should I add the `api_key` section back in too?

Comment (mdibaiee) [reply airbyte-integrations/connectors/source-airtable/spec.patch.json:9]: @jman0119 yes

File: airbyte-integrations/connectors/source-airtable/oauth2.patch.json (hunk 4 above)
Comment (mdibaiee) [reply airbyte-integrations/connectors/source-airtable/oauth2.patch.json:7]: @jman0119 yes I think that's a typo from me, `token_expiry_date` 👍🏽

**Comment by jman0119** [comment]
@mdibaiee thanks I made the changes and committed another PR! (I think I accidentally added the notion connector change to a commit on this ticket, not sure how to undo that... but that change was already merged as part of the other ticket for notion connector.)

**Pull request merged**